Approved/implicitly-scoped-qubit-allocation.md

```diff
@@ -0,0 +1,314 @@
+This syntax is preserved by this proposal with only minor changes.
+
+## Proposed Modification
+
+The new `use` and `borrow` statements behave like the existing `using` and `borrowing` statements with two differences:
+
+1. The block is optional.
+   If no block is provided, the scope of the qubit is implicitly defined to be from the `use` or `borrow` statement until the end of the current block.
+2. Parentheses around the header are not required.
+   For consistency, parentheses are no longer required around any block statement header.
+
+Each example below is equivalent to the corresponding example in the [Current Status](#current-status) section with the same number.
+
+### Examples
+
+Example 1b: A single qubit allocated and released at the same time as the end of the outer block.
+
+```qsharp
+operation FlipCoin() : Result {
+    use q = Qubit();
+    H(q);
+    return M(q); // q is released here.
+}
+```
+
+Example 2b: Many qubits allocated in a single `use` statement with tuple destructuring.
+
+```qsharp
+operation QubitTuple(n : Int) : Result {
+    use (a, b, c) = (Qubit[2 * n + 1], Qubit[n], Qubit());
+    // ...
+    return M(c); // a, b, and c are released here.
+}
+```
+
+Example 3b: Many qubits allocated with a separate `use` statement for each variable.
+
+```qsharp
+operation NestedBlocks(n : Int) : Result {
+    use a = Qubit[2 * n + 1];
+    use b = Qubit[n];
+    use c = Qubit();
+    // ...
+    return M(c); // a, b, and c are released here.
+}
+```
+
+Example 5b: A clean qubit and a borrowed qubit. This can only be expressed with multiple statements, not with tuple destructuring.
+
+```qsharp
+operation UsingAndBorrowing() : Result {
+    use a = Qubit();
+    borrow b = Qubit();
+    // ...
+    return M(a); // a and b are released here.
+}
+```
+
+Example 5b: A qubit allocated within a new block and released before the end of the outer block.
+
+```qsharp
+operation DifferentLifetime() : Result {
+    mutable r = Zero;
+    use q = Qubit() {
+        // Apply operations...
+        set r = M(q);
+        // q is released here.
+    }
+    // Classical post-processing...
+    return r;
+}
+```
+
+# Implementation
+
+The `use` and `borrow` keywords can be added, and the `using` and `borrowing` keywords can be deprecated.
+Until `using` and `borrowing` are removed, they can be used in place of `use` and `borrow` for both the block and non-block statements.
+
+The block for the existing `using` and `borrowing` syntax nodes can be made optional.
+Either the block remains optional throughout all stages of the syntax tree, or the non-block form is transformed into the block form by automatically moving all statements below it into a new block.
+In the former case, all syntax tree consumers need to be aware of both forms, while in the latter case consumers would notice no change from the current syntax tree node.
```

Comment:
I'd rather go with the latter. Do you see any issue with that?

Reply:
No, I think that could work, but I left it open-ended here in case things change during implementation.

Reply:
In general, if we introduce new items into the syntax tree (without a compiler step to eliminate it if desired), then we force every extension and backend that deals with the syntax tree to do work. Hence it is good to have clarity around how much work we impose on whom. :) I would clarify that here. 
In terms of process, it is of course possible that we run into unexpected issues with how we envisioned the implementation. That why we open an issue to track the status of the implementation when it starts and to discuss precisely if and how the proposal needs to be adapted. Having it stated explicitly here hence doesn't mean that we can't revise it during implementation, but it means there should be a good reason to revise it.